### ISSUE TITLE
Remove input validation for `voice`

### ISSUE THREAD

**Issue by Sidd065** [Lex-au/Orpheus-FastAPI#31]
[Multilingual models](https://canopylabs.ai/releases/orpheus_can_speak_any_language) provide these voices:
| Language | Voices                         | Supported Tags                                                                 |
|----------|--------------------------------|---------------------------------------------------------------------------------|
| French   | pierre, amelie, marie          | chuckle, cough, gasp, groan, laugh, sigh, sniffle, whimper, yawn               |
| German   | jana, thomas, max              | chuckle, cough, gasp, groan, laugh, sigh, sniffle, yawn                        |
| Korean   | 유나, 준서                      | 한숨, 헐, 헛기침, 훌쩍, 하품, 낄낄, 신음, 작은 웃음, 기침, 으르렁                     |
| Hindi    | ऋतिका (more coming)            | coming soon                                                                    |
| Mandarin | 长乐, 白芷                      | 嬉笑, 轻笑, 呻吟, 大笑, 咳嗽, 抽鼻子, 咳                                             |
| Spanish  | javi, sergio, maria            | groan, chuckle, gasp, resoplido, laugh, yawn, cough                            |
| Italian  | pietro, giulia, carlo          | sigh, laugh, cough, sniffle, groan, yawn, gemito, gasp                         |

The current implementation has a hard coded list of voices which could be updated with these voices or removed entirely.

**Comment by Lex-au** [comment]
Consolidating this into #29 — all discussion and resolution will continue there.

**Comment by PumaAI487** [comment]
Asking here because of the merge. I did see your comment Lex-au, just wondering since I know the voice cloning itself is not working for now, but if we have already trained voices from voice-models dot com, with two files being a, ".pt" format, and ".index" file, if there was a folder location to put those files to use them, and how to update the files below? I did trying searching the project folder for where they might be but only reference to the voices themselves are in the "inference.py" and in the templates "tts.html" file.

Adding New Voices
To add new voices, update the AVAILABLE_VOICES list in tts_engine/inference.py and add corresponding descriptions in the HTML template.

Inference.py @ line 45 
"# Available voices based on the Orpheus-TTS repository
AVAILABLE_VOICES = ["tara", "leah", "jess", "leo", "dan", "mia", "zac", "zoe"]
DEFAULT_VOICE = "tara"  # Best voice according to documentation"

tts.html @ line 167
   <div class="text-xs text-dark-300">
                      {% if voice_option == "tara" %}Female, conversational, clear
                      {% elif voice_option == "leah" %}Female, warm, gentle
                      {% elif voice_option == "jess" %}Female, energetic, youthful
                      {% elif voice_option == "leo" %}Male, authoritative, deep
                      {% elif voice_option == "dan" %}Male, friendly, casual
                      {% elif voice_option == "mia" %}Female, professional, articulate
                      {% elif voice_option == "zac" %}Male, enthusiastic, dynamic
                      {% elif voice_option == "zoe" %}Female, calm, soothing
                      {% endif %}
                    </div>

Or do I have it wrong and only voices that can be added are ones created/added by your team? Since I don't believe the ones off that website have emotion tagging builtin or added to them?

Sorry if this is a dumb question lol

**Comment by Lex-au** [comment]
If you're not keen to wait for my full update this weekend—good news, you're already on the right track.

To add new voice actors to your Orpheus checkpoint, head to `inference.py`. Under the `AVAILABLE_VOICES` list, simply add your new entries. Just make sure each name you add is **callable by the model** you're using.

For example, if you're using the **French checkpoint**, update the list like this:

```python
AVAILABLE_VOICES = ["tara", "leah", "jess", "leo", "dan", "mia", "zac", "zoe", "pierre", "amelie", "marie"]
```

Optional: Updating the Frontend (Web UI)

If you're using the **Web UI** and want your new voices to show up with proper labels in `tts.html`, update the descriptive block like so:

```html
<div class="text-xs text-dark-300">
  {% if voice_option == "tara" %}Female, conversational, clear
  {% elif voice_option == "leah" %}Female, warm, gentle
  {% elif voice_option == "jess" %}Female, energetic, youthful
  {% elif voice_option == "leo" %}Male, authoritative, deep
  {% elif voice_option == "dan" %}Male, friendly, casual
  {% elif voice_option == "mia" %}Female, professional, articulate
  {% elif voice_option == "zac" %}Male, enthusiastic, dynamic
  {% elif voice_option == "zoe" %}Female, calm, soothing
  {% elif voice_option == "pierre" %}Male, French, soft
  {% elif voice_option == "amelie" %}Female, French, youthful
  {% elif voice_option == "marie" %}Female, French, casual
  {% endif %}
</div>
```

If you're using **only the API**, you can skip this step entirely.

**Comment by Lex-au** [comment]
> Correction meant ".pth" and ".index"
> 
> I have the local installation of Orpheustts that use the Webui with LMstudio for using the Orpheustts 4b model. More so how would I go about making voices not provided by your team such as tara, or your new updated list of voices such as "marie" etc but my own locally stored trained voices? Or is it not possible?

To create your own voice actors based on the existing model, you would need to fine-tune a new checkpoint with your custom voice data. As of now, I haven’t personally explored voice cloning workflows specifically for OrpheusTTS, but the developers have indicated that it *is* possible.

So in short: yes, but it requires some additional training steps—it's not just a matter of plugging in a `.pth` and `.index` file unless those represent a model already fine-tuned to include your custom voice actor.

**Comment by PumaAI487** [comment]
Oh okay that makes more sense

**Comment by Lex-au** [comment]
> Oh okay that makes more sense

I'll look into this more closely over the weekend. If the developers have provided any detailed guidance on the fine-tuning process—particularly how it interacts with the model architecture and tokenizer—I’ll see if I can turn it into a streamlined application or utility for creating custom voice actors.

**Comment by PumaAI487** [comment]
> > Oh okay that makes more sense
> 
> I'll look into this more closely over the weekend. If the developers have provided any detailed guidance on the fine-tuning process—particularly how it interacts with the model architecture and tokenizer—I’ll see if I can turn it into a streamlined application or utility for creating custom voice actors.

That'd be awesome if they do, and you're able to. New to all this, have yet to train an llm, let alone voice model, i was thinking the ones one voice-model website that say they were trained were somewhat drag and drop but very over my head. 

Appreciate the replies, and hope you have a great easter weekend.

**Comment by Lex-au** [comment]
Hi @PumaAI487,

Hope you had a good Easter! I took some time over the break to dig into this, and I now have a solid grasp on how to add new voice actors into the Orpheus model.

For example, I distilled a version of [Sesame’s Maya](https://huggingface.co/models?search=lex-au%20kaya) and fine-tuned it into the Orpheus base pretrained model. Afterward, I quantised it. The training was pretty lightweight—just 49 audio clips, trained for a single epoch over ~200 steps. The whole process took about 15 minutes on my 4090.

The results? Meh—passable, but not production-quality. If you scale up to 100+ clips with varied durations (say, 5–20 seconds each), you’d start seeing quality comparable to—or even better than—the default voice like Tara.

To help others replicate the process, I published an example of how the [dataset](https://huggingface.co/datasets/lex-au/Orpheus-3b-Kaya) should be structured before creating a LoRA and merging the final checkpoint. Honestly, I think the devs at Canopy left out some *very* critical steps in their documentation—whether intentionally or not is up for debate. They’re a small team, so I half-jokingly assume they let the intern write the training guide.

It’s definitely a complex workflow, but it’s doable with some patience.

I’m considering turning this into its own project—something that would guide users through distillation, fine-tuning, merging, quantising, and even auto-uploading their models to Hugging Face or ModelScope. Might be able to release that sometime later this month.

**Comment by PumaAI487** [comment]
Yeah it was good, hope yours was aswell!

The process is a bit over my head for now but laying out the basic steps like that gives me a path I could try to follow. That's awesome, really great work! I think people would love a guide I know I would. 

I have plenty of questions but I'll try figuring out some of those basics (distill, fine tune, quantizing, and merging voice model lora with I believe the Orpheus 3B q4_k_m model, get a better understanding of the process before I throw a truckload of questions your way aha.

Yeah open source stuff I never get mad at the devs, and dislike people that do, not like they paid them. And I appreciate you taking the time during holiday to figure this out!
